packages/stryker-mocha-framework/src/MochaTestFramework.ts

```diff
@@ -14,22 +14,24 @@ export default class MochaTestFramework implements TestFramework {
     });`;
   }
 
-  filter(testIds: number[]) {
-    return `
-      var mocha = window.mocha || require('mocha');
+  filter(testSelections: TestSelection[]) {
+    const selectedTestNames = testSelections.map(selection => selection.name);
+    return `var Mocha = window.Mocha || require('mocha');
+      var selectedTestNames = ${JSON.stringify(selectedTestNames)};
       if (window.____mochaAddTest) {
-        mocha.Suite.prototype.addTest = window.____mochaAddTest;
+        Mocha.Suite.prototype.addTest = window.____mochaAddTest;
       } else {
-        window.____mochaAddTest = mocha.Suite.prototype.addTest
+        window.____mochaAddTest = Mocha.Suite.prototype.addTest
       }
       var current = 0;
-      var realAddTest = mocha.Suite.prototype.addTest;
-      mocha.Suite.prototype.addTest = function () {
-        if (${JSON.stringify(testIds)}.indexOf(current) > -1) {
+      var realAddTest = Mocha.Suite.prototype.addTest;
+      
+      Mocha.Suite.prototype.addTest = function (test) {
+        // Only actually add the tests with the expected names
+        var name = this.fullTitle() + ' ' + test.title;
+        if(selectedTestNames.indexOf(name) !== -1) {
           realAddTest.apply(this, arguments);
         }
-        current++;
```

Comment:
You've removed this line but the declaration is still there. We can remove it right?

Reply:
Good point. I'll remove it.